content/articles/statistical-hypothesis-testing-python-implementation/index.md

```diff
@@ -0,0 +1,573 @@
+The three methods to be discussed are:
+1. Chi-square test
+2. T-test
+3. ANOVA test
+ 
+#### Chi-square test
+Chi-square test is used to analyze two categorical variables in a data set. It focuses on the relationship between these two categorical variables. For example, comparing the after effects of malaria drug A to those of malaria drug B.
+
+Chi-square is a tool for checking how divergent the observed frequency is from the expected results.
+
+In a Chi-square test, we accept the null hypothesis when variable A and variable B are not dependent or have no relationship between them.
+
+We also use the alternate hypothesis when variable A depends on variable B. 
+
+The formula for calculating Chi-square test is shown below:
+ 
+![chi_square_formula](/engineering-education/statistical-hypothesis-testing-python-implementation/chi-square.png)
+
+Consider the following example:
+ 
+A mathematics teacher conducted a test for a class of 20 students. He expected 5 to score good marks, 7 to score average marks while 8 to fail. Eventually, when the result came out, 5 failed, 6 were average while 9 scored good marks. 
+
+We can use a Chi-square test to analyze the performance of this class.
+ 
+|  | Good  |  Average  | Fail | Total |
+|----|--------|:----------|--------------|--------|
+| Observed| 9 | 6 | 5 | 20 |
+| Expected| 5 | 7 | 8 | 20 |
+
+From the table above, students fall into three categories ranging from what is expected to what is finally recorded. The main categories are `good`, `average`, and `fail`. 
+
+Let's denote the expected and observed values with letters to make it more understandable.
+
+Let the: 
+- Observed good be `A` and the expected good be `a`.
+- Observed average be `B` and the expected average be `b`.
+- Observed fail be `C` and the expected to fail be `c`.
+
+We then find the difference between the expected and observed values, as shown below:
+  
+```
+A-a = 4
+B-b = -1
+C-c = -3
+```
+
+```
+/A-a/² = 16
+/B-b/² = 1
+/C-c/² = 9
+```
+```
+Good = (16/5) = 3.2
+Average = (1/7) = 0.143
+Fail = (9/8) = 1.125
+ 
+Chi-square = 3.2 + 0.143 + 1.125
+Chi-square = 4.468
+```
+For huge datasets, mathematical calculations cannot be done by hand. The implementation of the Chi-square test in Python is discussed in the next section.
+
+#### Chi-square test with Python implementation.
+We need to import `pandas`, `numpy`, `scipy`and , as demonstrated below:
+ 
+```py
+import pandas as pd
+import numpy as np
+import scipy.stats as stats
+from scipy.stats import f_oneway
+from scipy.stats import chi2_contingency
+import matplotlib.pyplot as plt
+ 
+import statsmodels.api as sm
+from statsmodels.formula.api import ols
+ 
+import warnings
+warnings.filterwarnings( "ignore" )
+```
+ 
+The libraries imported are the most important libraries used for any data analysis project.
+
+`f_oneway` and `chi2_contigency` libraries are used for implementing a one-way `ANOVA` and Chi-square tests respectively.
```

Comment:
"ANOVA" need not be put in quotes at all. And please change "Chi-square" to "chi-square" (small c)
Change that in the rest of the draft.